### PR TITLE
magick.so: explicitly reference wand/core libraries

### PR DESCRIPTION
**Pull request by traeak**
Fixes the magick plugin linking for gcc ubuntu 20.04 and debian 11

### CODE REVIEW

**Comment by bneradt** [comment]
[approve ci autest]

File: plugins/experimental/magick/CMakeLists.txt [hunk 1]
```diff
@@ -16,6 +16,11 @@
 #######################
 
 add_atsplugin(magick magick.cc)
-target_link_libraries(magick PRIVATE ImageMagick::Magick++ ts::tscppapi OpenSSL::Crypto)
+
+if(DEFINED ImageMagic::MagickWand)
```
Comment (cmcfarlen) [inline plugins/experimental/magick/CMakeLists.txt:20]: You can use TARGET here to help readability.

**Review by cmcfarlen — CHANGES_REQUESTED** [review]

File: plugins/experimental/magick/CMakeLists.txt (hunk 1 above)
Comment (traeak) [reply plugins/experimental/magick/CMakeLists.txt]: removed the if checks.

**Comment by traeak** [comment]
[approve ci autest]

**Review by cmcfarlen — APPROVED** [review]

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**